### PR TITLE
LP-2712 Display business line courses in course catalog

### PR DESCRIPTION
**Pull request by asadali145**

### CODE REVIEW

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/1969/workflows/workflows_id/jobs/1969/artifacts)**. Coverage !!! :scream:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/managers.py (83.3%): Missing lines 38,60
openedx/adg/lms/courseware_override/helpers.py (25.0%): Missing lines 31-32,34
-------------
Total:   16 lines
Missing: 5 lines
Coverage: 68%
-------------
```

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/1971/workflows/workflows_id/jobs/1971/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/managers.py (100%)
openedx/adg/lms/conftest.py (100%)
openedx/adg/lms/courseware_override/helpers.py (75.0%): Missing lines 32
openedx/adg/lms/courseware_override/tests/test_helpers.py (100%)
openedx/adg/lms/webinars/tests/test_api_views.py (100%)
-------------
Total:   34 lines
Missing: 1 line
Coverage: 97%
-------------
```

File: openedx/adg/lms/applications/managers.py [hunk 1]
```diff
@@ -23,29 +23,47 @@ class MultilingualCourseGroupManager(Manager):
...
         """
         return self.get_queryset().filter(is_program_prerequisite=True, multilingual_courses__isnull=False).distinct()
 
-    def get_courses(self, user, is_prereq=False):
+    def business_line_course_groups(self, user):
+        return self.get_queryset().filter(
+            Q(is_common_business_line_prerequisite=True) | Q(business_line_prerequisite=user.application.business_line)
+        )
+
+    def get_courses(self, user, is_prereq=False, business_line_courses=False):
```
Comment (ahmed-faraz46) [inline openedx/adg/lms/applications/managers.py:42]: @asadali145 here are terminologies we are currently using for different kind of pr reqs
- Program pre reqs (previously known as per reqs)
- Business line pre req (pre reqs of a specific business line)
- Common business line pre req (Pre reqs which are common for all of the business lines)

Let's please update our variable names at all of the places accordingly, in order to enhance the readability.

File: openedx/adg/lms/applications/managers.py [hunk 2]
```diff
@@ -23,29 +23,47 @@ class MultilingualCourseGroupManager(Manager):
...
-        course_groups = self.prereq_course_groups() if is_prereq else self.get_queryset()
+        course_groups = self.common_course_groups()
+
+        if is_prereq:
+            course_groups = self.prereq_course_groups()
+
+        if business_line_courses:
+            course_groups = self.business_line_course_groups(user)
 
+        courses_list = []
```
Comment (ahmed-faraz46) [inline openedx/adg/lms/applications/managers.py:62]: How about writing separate getters for all these three kind of course groups and moving the code below to a utility and utilizing in all three getters?

**Review by ahmed-faraz46 — CHANGES_REQUESTED** [review]

**Review by javeriaejaz-arbisoft — APPROVED** [review]
LGTM

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/2001/workflows/workflows_id/jobs/2001/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/helpers.py (100%)
openedx/adg/lms/applications/management/commands/update_is_prerequisite_courses_passed.py (100%)
openedx/adg/lms/applications/managers.py (100%)
openedx/adg/lms/applications/models.py (100%)
openedx/adg/lms/applications/tests/test_helpers.py (100%)
openedx/adg/lms/applications/tests/test_models.py (100%)
openedx/adg/lms/applications/views.py (100%)
openedx/adg/lms/conftest.py (100%)
openedx/adg/lms/courseware_override/helpers.py (42.9%): Missing lines 18-19,21,39
openedx/adg/lms/courseware_override/tests/test_helpers.py (100%)
openedx/adg/lms/webinars/tests/test_api_views.py (100%)
-------------
Total:   71 lines
Missing: 4 lines
Coverage: 94%
-------------
```

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/2002/workflows/workflows_id/jobs/2002/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/helpers.py (100%)
openedx/adg/lms/applications/management/commands/update_is_prerequisite_courses_passed.py (100%)
openedx/adg/lms/applications/managers.py (100%)
openedx/adg/lms/applications/models.py (100%)
openedx/adg/lms/applications/tests/test_helpers.py (100%)
openedx/adg/lms/applications/tests/test_models.py (100%)
openedx/adg/lms/applications/views.py (100%)
openedx/adg/lms/conftest.py (100%)
openedx/adg/lms/courseware_override/helpers.py (42.9%): Missing lines 18-19,21,39
openedx/adg/lms/courseware_override/tests/test_helpers.py (100%)
openedx/adg/lms/webinars/tests/test_api_views.py (100%)
-------------
Total:   71 lines
Missing: 4 lines
Coverage: 94%
-------------
```

**Review by annas-usman — APPROVED** [review]

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/2004/workflows/workflows_id/jobs/2004/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/helpers.py (100%)
openedx/adg/lms/applications/management/commands/update_is_prerequisite_courses_passed.py (100%)
openedx/adg/lms/applications/managers.py (100%)
openedx/adg/lms/applications/models.py (100%)
openedx/adg/lms/applications/tests/test_helpers.py (100%)
openedx/adg/lms/applications/tests/test_models.py (100%)
openedx/adg/lms/applications/views.py (100%)
openedx/adg/lms/conftest.py (100%)
openedx/adg/lms/courseware_override/helpers.py (42.9%): Missing lines 18-19,21,39
openedx/adg/lms/courseware_override/tests/test_helpers.py (100%)
openedx/adg/lms/webinars/tests/test_api_views.py (100%)
-------------
Total:   71 lines
Missing: 4 lines
Coverage: 94%
-------------
```

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/2005/workflows/workflows_id/jobs/2005/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/helpers.py (100%)
openedx/adg/lms/applications/management/commands/update_is_prerequisite_courses_passed.py (100%)
openedx/adg/lms/applications/managers.py (100%)
openedx/adg/lms/applications/models.py (100%)
openedx/adg/lms/applications/tests/test_helpers.py (100%)
openedx/adg/lms/applications/tests/test_models.py (100%)
openedx/adg/lms/applications/views.py (100%)
openedx/adg/lms/conftest.py (100%)
openedx/adg/lms/courseware_override/helpers.py (42.9%): Missing lines 18-19,21,39
openedx/adg/lms/courseware_override/tests/test_helpers.py (100%)
openedx/adg/lms/webinars/tests/test_api_views.py (100%)
-------------
Total:   71 lines
Missing: 4 lines
Coverage: 94%
-------------
```

**Review by ahmed-faraz46 — APPROVED** [review]

**Pull request merged**